File: src/main/java/com/github/zyypj/tadeuBooter/api/http/client/HttpResponse.java
package com.github.zyypj.tadeuBooter.api.http.client;

import lombok.Generated;

public class HttpResponse<T> {
    private final T rawData;
    private final String data;
    private final Integer statusCode;
    private final Long time;
    private final String headers;
    private final HttpStatus httpStatus;

    public HttpResponse(T rawData, String data, Integer statusCode, Long time, String headers, HttpStatus httpStatus) {
        this.rawData = rawData;
        this.data = data;
        this.statusCode = statusCode;
        this.time = time;
        this.headers = headers;
        this.httpStatus = httpStatus;
    }

    @Generated
    public T getRawData() {
        return this.rawData;
    }

    @Generated
    public String getData() {
        return this.data;
    }

    @Generated
    public Integer getStatusCode() {
        return this.statusCode;
    }
}

File: shared/src/main/java/me/zyypj/booter/shared/http/client/HttpResponse.kt
package me.zyypj.booter.shared.http.client


/**
 * Representa a resposta de uma requisição HTTP.
 *
 * @param rawData     dado bruto (tipado) obtido da requisição – útil quando você deseja
 *                    trabalhar com JSON já desserializado, por exemplo.
 * @param data        corpo em texto puro.
 * @param statusCode  código de status HTTP.
 * @param time        tempo de resposta em milissegundos.
 * @param headers     cabeçalhos em string.
 * @param httpStatus  enum simplificado mapeado pelo código.
 */
data class HttpResponse<T>(
    val rawData: T,
    val data: String,
    val statusCode: Int,
    val time: Long,
    val headers: String,
    val httpStatus: HttpStatus,
)